src/PRASBase/assets.jl

```diff
@@ -146,6 +146,60 @@ Base.:(==)(x::T, y::T) where {T <: Storages} =
     x.λ == y.λ &&
     x.μ == y.μ
 
+Base.getindex(s::S, idxs::AbstractVector{Int}) where {S <: Storages} =
+    S(s.names[idxs], s.categories[idxs],s.charge_capacity[idxs,:],
+      s.discharge_capacity[idxs, :],s.energy_capacity[idxs, :],
+      s.charge_efficiency[idxs, :], s.discharge_efficiency[idxs, :], 
+      s.carryover_efficiency[idxs, :],s.λ[idxs, :], s.μ[idxs, :])
+
+function Base.vcat(stors::S...) where {N, L, T, P, E, S <: Storages{N,L,T,P,E}}
```

Comment:
This is more of a complaint to my past self (who wrote the `Generators` version of this), but in the spirit of continuous improvement, I don't think there's any reason this (and the equivalents for the other resource types) couldn't just be `Base.vcat(stors::Storages{N,L,T,P,E}...) where {N, L, T, P, E}` instead, which would be much easier to understand while still extracting the needed inner type parameters. (The logic for `getindex` is different though, since there we're calling `S` as a constructor.) Or am I missing something?

If not, please go ahead and fix this in the `Generators` version of these functions as well.

Reply:
No, this makes sense. I will fix it.